plugins/inventory/windows_azure.py

```diff
@@ -82,15 +90,39 @@ def __init__(self):
 
         if self.args.list_images:
             data_to_print = self.json_format_dict(self.get_images(), True)
-        elif self.args.list:
+        elif self.args.list or self.args.host:
             # Display list of nodes for inventory
             if len(self.inventory) == 0:
-                data_to_print = self.get_inventory_from_cache()
+                data = json.loads(self.get_inventory_from_cache())
             else:
-                data_to_print = self.json_format_dict(self.inventory, True)
+                data = self.inventory
 
+            if self.args.host:
+                data_to_print = self.get_host(self.args.host)
+            else:
+                # Add the `['_meta']['hostvars']` information.
+                hostvars = {}
+                if len(data) > 0:
+                    for host in set(reduce(lambda x, y: x + y,
+                                           [i for i in data.values()])):
```

Comment:
`data.values()` is a list of lists, so it's just flattening that into one list.  I've updated it to use a nested list comprehension instead, which may be (slightly) more readable.